### PR TITLE
feat: enable `docs/v8.x`

### PR DESCRIPTION
**Pull request by mdjermanovic**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request?

Updates the redirects file to enable `/docs/v8.x/`, where we'll have v8.57.0 docs.

#### What changes did you make? (Give an overview)

Copy-pasted and adjusted the code that handles `/docs/head/`.

#### Related Issues

https://github.com/eslint/eslint/issues/18229

### CODE REVIEW

**Comment by mdjermanovic** [comment]
What to check: https://deploy-preview-541--new-eslint.netlify.app/docs/v8.x/ should look normal, except that the version in selectors is empty as this is not an "officially" published docs site yet. When we release the v9.0.0 final tomorrow, v8.57.0 will be selected.

File: src/_data/sites/en.yml
Comment (mdjermanovic) [inline src/_data/sites/en.yml:23]: Because of the dot in the branch name, the subdomain on Netlify is `v8-x--docs-eslint.netlify.app`, not `v8.x--docs-eslint.netlify.app`.

**Comment by mdjermanovic** [comment]
I think it's okay to merge this once approved (i.e., I think merging doesn't have to wait for the v9.0.0 release) so that we can check it live on eslint.org.

**Review by nzakas — APPROVED** [review]

**Pull request merged**
